tools/deploy_image_service.py

```diff
@@ -13,15 +14,46 @@
 SRC_FILE = os.path.join(os.getcwd(), 'deploy/assisted-image-service.yaml')
 DST_FILE = os.path.join(os.getcwd(), 'build', deploy_options.namespace, 'assisted-image-service.yaml')
 
+SERVICE_SRC_FILE = os.path.join(os.getcwd(), 'deploy/assisted-image-service-service.yaml')
+SERVICE_DST_FILE = os.path.join(os.getcwd(), 'build', deploy_options.namespace, 'assisted-image-service-service.yaml')
 
 def main():
     utils.verify_build_directory(deploy_options.namespace)
 
+    # Handle assisted-image-service `service`
+    with open(SERVICE_SRC_FILE, "r") as src:
+        with open(SERVICE_DST_FILE, "w+") as dst:
+            raw_data = src.read()
+            raw_data = raw_data.replace('REPLACE_NAMESPACE', f'"{deploy_options.namespace}"')
+            data = yaml.safe_load(raw_data)
+
+            print("Deploying {}".format(SERVICE_DST_FILE))
+            dst.write(yaml.dump(data))
+
+    if deploy_options.apply_manifest:
+        utils.apply(
+            target=deploy_options.target,
+            namespace=deploy_options.namespace,
+            file=SERVICE_DST_FILE
+        )
+
+    # Handle assisted-image-service `deployment`
     with open(SRC_FILE, "r") as src:
         log.info(f"Loading source template file for assisted-image-service: {SRC_FILE}")
         raw_data = src.read()
         raw_data = raw_data.replace('REPLACE_NAMESPACE', f'"{deploy_options.namespace}"')
         raw_data = raw_data.replace('REPLACE_IMAGE_SERVICE_IMAGE', os.environ.get("IMAGE_SERVICE"))
+
+        image_service_base_url = os.environ.get("IMAGE_SERVICE_BASE_URL", utils.get_service_url(service="assisted-image-service",
```

Comment:
Why put this override env var here?
Shouldn't the returned url always be correct?

Reply:
It's needed for test-infra, as IMAGE_SERVICE_BASE_URL is set in the deploy [script](https://github.com/openshift/assisted-test-infra/blob/8bd7be1e186f8beed03c76eff52e24392d6647bb/scripts/deploy_assisted_service.sh#L13) (to <main_ip>:6016)

Reply:
Are we sure that's being used? In the script to deploy the configmap it's not reading this env var and only getting the value from the `get_service_url` call.

ref: https://github.com/openshift/assisted-service/blob/a6527c6c9e877f57ae66e67a9acb6599f265ae44/tools/deploy_assisted_installer_configmap.py#L71-L76

Reply:
Yeah but iiuc, the assisted-service cm is updated before by the test-infra script: https://github.com/openshift/assisted-test-infra/blob/bc9ae8737d5415e5eeae3bb328cc3087351d69df/scripts/deploy_assisted_service.sh#L131

Reply:
Ah okay, that makes sense. Can you add a comment to explain why it's here?
It's a bit confusing since it's different than what we're doing for assisted-service.